doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+
+Specifying common configuration across all collections
+------------------------------------------------------
+
+If there are properties you want to apply across all collections, those can be supplied in ``hook_parameters`` within the ``DebuggerHookConfig`` object. For example, to apply a value of ``save_interval`` across all collections do:
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config_1 = CollectionConfig(
+        name='collection_name_1',
+        parameters={
+            "include_regex": ".*"
+        }
+    )
+    collection_config_2 = CollectionConfig(
+        name='collection_name_2',
+        parameters={
+            "include_regex": ".*"
+        }
+    }
+
+    debugger_hook_config = DebuggerHookConfig(
+        s3_output_path='s3://path/for/data/emission',
+        container_local_output_path='/local/path/for/data/emission'
+        hook_parameters={
+            "save_interval": "10"
+        },
+        collection_configs=[
+            collection_config_1, collection_config_2
+        ]
+    )
+
+In the above sample code, the ``save_interval`` of ``10`` will be applied for storing both collections.
+
+Note that the value set in the ``collection_parameters`` for a parameter will override the corresponding value of the same parameter in the ``hook_parameters``. For example, in the above sample code, if ``collection_config_2`` had the value of ``save_interval`` set something other than ``10``, say, ``20``, then the tensors for that collection would have been saved with step interval ``20`` while those for ``collection_config_1`` would still be saved with ``10``.
+
+Continuous analyses through Rules
+=================================
+
+In addition to collecting the debugging data, Amazon SageMaker Debugger provides the capability for you to analyze it in a streaming fashion through "rules". SageMaker Debugger 'Rules' are pieces of code which encapsulate the logic for analyzing the debugging data.
+
+SageMaker Debugger provides a set of built-in rules curated by data scientists and engineers at Amazon. There is additional support for bringing in custom rule source codes for evaluation. In the following sections, you'll learn how to use either while training your model.
+
+Relationship between debugger hook and rules
+--------------------------------------------
+
+Using SageMaker Debugger is, broadly, a two-pronged approach. One one hand you have production of debugging data, which is done through the Debugger Hook and on the other hand you have the consumers which can be rules (for continuous analyses) or SageMaker Debugger SDK (for interactive analyses).
```

Comment:
s/One/On

also add a comma after "Hook"

Reply:
Seems to me this point could usefully be make earlier.

Reply:
* [@laurenyu ] Fixed. Thanks.
* [@aws-rbs ] I'd rather explain it here since the concept of rules (in particular usage through Python SDK) wasn't explained earlier and can potentially leave the reader searching for context.